### PR TITLE
Fix propagation of EFFECTIVE_VERSION during builds in pipeline

### LINKED ISSUE #47: Installation image is using wrong tag by default
**What happened**:
 The installation container cannot be started due to wrong default image tag
```  
Normal  BackOff  4m20s (x661 over 153m)  kubelet  Back-off pulling image "eu.gcr.io/gardener-project/gardener/extensions/runtime-gvisor-installation:v0.5.1-dev"
```

**What you expected to happen**:
 A proper image tag to be used by default.

**How to reproduce it (as minimally and precisely as possible)**:

Create a shoot cluster with worker pool with gVisor enabled. With the following controller-registration.yaml: https://raw.githubusercontent.com/gardener/gardener-extension-runtime-gvisor/v0.5.1/example/controller-registration.yaml

**Anything else we need to know?**:
The issue lies (as far as we found out) in the gvisor-runtime image https://console.cloud.google.com/gcr/images/gardener-project/eu/gardener%2Fextensions%2Fruntime-gvisor@sha256:866ea708c0ce5e5c64b9dc1a290b89caa90151691253d328af4155cdf4e64f3f which creates a daemonset with a "-dev" image tag. We had a look at the included binary and found the string "v0.5.1-dev". If we build the image locally (with v0.5.1 checked out) this string is "v0.5.1". So we assume that a wrong version is linked into the upstream image. Currently we assume an issue in the build pipeline but unfortunately we cant get the log from the respective run: https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-extension-runtime-gvisor-release-v0.5/jobs/release-v0.5-release-job/builds/1

**Environment**:

- Gardener version : 1.46
- Extension version: 0.5.1
- Kubernetes version (use `kubectl version`): 1.23
- Cloud provider or hardware configuration: 
- Others:

### PR DESCRIPTION
**Pull request by vpnachev**
**What this PR does / why we need it**:
Fix propagation of EFFECTIVE_VERSION during builds in pipeline.
Also splits docker builder stage to speed-up image builds.

**Which issue(s) this PR fixes**:
Fixes #47 

**Special notes for your reviewer**:
/invite @ialidzhikov 

**Release note**:

```bugfix developer
A bug that non-released version of the extension was using wrong version of the installation image when no image vector overwrite is configured is fixed.
```

### CODE REVIEW

**Review by ialidzhikov — COMMENTED** [review]
Are you sure that it is that trivial? In the past you did a similar change in https://github.com/gardener/gardener-extension-runtime-gvisor/pull/36 and then you reverted in https://github.com/gardener/gardener-extension-runtime-gvisor/pull/39.

**Comment by vpnachev** [comment]
Good catch.
/hold

**Comment by vpnachev** [comment]
After https://github.com/gardener/cc-utils/commit/2348bceedd0b9bd49ba813fe0cc708f0ab557fe4, the `EFFECTIVE_VERSION` is set as build arg and the `GitVersion` is properly set.
```bash
$ docker run  --rm eu.gcr.io/gardener-project/gardener/extensions/runtime-gvisor:v0.7.0-dev-c94906ece86738c32a4558d2dba378aaa289837f-linux-amd64 --version=raw | grep -Po 'GitVersion:".*?"'
GitVersion:"v0.7.0-dev-c94906ece86738c32a4558d2dba378aaa289837f"
```

/title Fix propagation of EFFECTIVE_VERSION during builds in pipeline
/unhold

**Comment by vpnachev** [comment]
This arm64 build seems to take forever most of the time. It should be because no mater of the target, always the golang binaries are built or the gVisor binaries are pulled. With [fdb8413](https://github.com/gardener/gardener-extension-runtime-gvisor/pull/62/commits/fdb8413ede2a5f60a467b3550ad8085ed77c8dd8), I've split the builder stage to avoid unnecessary build steps. Let's see if it will really improve.

**Comment by vpnachev** [comment]
> Let's see if it will really improve.

For the amd64 builds there is already an improvement:
- extension image build time is reduced from 90 to 60 seconds
- installation image build time reduced from 90 to 14 seconds (320 to 30 seconds for arm64)

Unfortunately, it seems the arm64 builds to get stuck.

**Review by ialidzhikov — COMMENTED** [review]
Minor suggestion, otherwise lgtm.

I can confirm that the effective version is correctly injected in the controller image/binary.

File: Makefile [hunk 1]
```diff
@@ -50,20 +55,34 @@ include vendor/github.com/gardener/gardener/hack/tools.mk
 .PHONY: install
 install:
 	@LD_FLAGS="$(LD_FLAGS)" \
+	EFFECTIVE_VERSION=$(EFFECTIVE_VERSION) \
```
[the remark this answers is not in the document]
Comment (vpnachev) [reply Makefile]: Good catch, thanks!

**Review by ialidzhikov — APPROVED** [review]
/lgtm

Thanks for picking this ad-hoc item!

**Pull request merged**